uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/performance/DeviceMonitor.java

```diff
@@ -0,0 +1,219 @@
+            return capitalize(model);
+        } else {
+            return capitalize(manufacturer) + " " + model;
+        }
+    }
+
+
+    private static String capitalize(String s) {
+        if (s == null || s.length() == 0) {
+            return "";
+        }
+        char first = s.charAt(0);
+        if (Character.isUpperCase(first)) {
+            return s;
+        } else {
+            return Character.toUpperCase(first) + s.substring(1);
+        }
+    }
+
+    /**
+     * Gets the current application's uid
+     *
+     * @return the application uid
+     */
+    public static int getApplicationUid() {
+        final Pattern pattern = Pattern.compile("userId=(\\d+)");
+        final String command = "dumpsys package " + packageName;
+
+        final Matcher matcher = pattern.matcher(AdbShellUtils.executeShellCommand(command));
+
+        if (matcher.find()) {
+            return Integer.parseInt(Objects.requireNonNull(matcher.group(1)));
+        }
+        return 0;
+    }
+
+    /**
+     * Get the total System memory
+     *
+     * @return
+     */
+    public static Long getTotalMemory() {
+        return processInfo.getTotalSystemMemory();
+    }
+
+    /**
+     * Get the current CPU usage
+     *
+     * @return the current cpu usage as a percentage e.g. 3.3%
+     */
+    public static Double getCpuUsage() {
+        return getStats(PerformanceProfile.CPU, Double.class);
+    }
+
+    /**
+     * Get the current memory usage
+     *
+     * @return the current memory usage in KB
+     */
+    public static Long getMemoryUsage() {
+        return getStats(PerformanceProfile.MEMORY, Long.class);
+    }
+
+    /**
+     * Get the current network info
+     *
+     * @return the information regarding the network usage
+     */
+    public static NetworkUsageMonitor.TrafficInfo getNetworkTrafficInfo() {
+        return getStats(PerformanceProfile.NETWORK, NetworkUsageMonitor.TrafficInfo.class);
+    }
+
+    /**
+     * Get the statistics of a particular performance profile
+     *
+     * @param performanceProfile the performance profile
+     * @param classOfT           the class of the stats result
+     * @return the current data showing its usage.
+     */
+    public static <T> T getStats(PerformanceProfile performanceProfile, Class<T> classOfT) {
+        PerformanceProfileMonitor<?> monitor = profilers.get(performanceProfile);
+
+        if (monitor == null) {
+            throw new IllegalArgumentException("The performanceProfile does not have a monitor.");
+        }
+
+        loadProcessInfo();
+
+        return classOfT.cast(monitor.getStats(processInfo));
+    }
+
+    /**
+     * Fetch the current process information
+     *
+     * @return the current process information
+     */
+    public static ProcessInfo getProcessInfo() {
+        return processInfo;
+    }
+
+    /**
+     * Updates or adds a new profiling monitor for a particular profile
```

Comment:
```suggestion
     * Updates or adds a new profiling monitor for a particular profile.
```

Reply:
Please add the `{@link <type>}` for both `PerformanceProfile` as well as `PerformanceProfileMonitor` classes in the javadoc here and everywhere else as applicable